### PR TITLE
#166272328 setup unit testing for project

### PR DESCRIPTION
**Pull request by emmygozi**
#### What does this PR do?

This Pull Request sets up a unit test for project and tests user register function. It also mocks helper methods like trim and error handler.

#### Description of Task to be completed?

- Sets up the unit test using chai and chai-http
- Tests user register function 
- Mocks trim helper method
- Mocks error handler methos

#### How should this be manually tested?

- `git clone https://github.com/andela/kingsmen-ah-backend.git` 

- `npm install` project dependencies

- Push project to github and click on Travis build

- Travis build fails beacause there is no test provided yet

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/166272328

#### Screenshots (if appropriate)
N/A
#### Questions:

N/A

### CODE REVIEW

**Comment by despeauxz** [comment]
Nice job! The build passes and nice descriptive test

**Review by despeauxz — APPROVED** [review]
Nice job! The build passes and nice descriptive test

**Review by fidelisojeah — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+node_js:
+- 10.15.3
+cache:
+  directories:
+  - node_modules
+before_install:
+- npm install
+before_script:
+  - psql -c 'CREATE DATABASE travis;' -U postgres
+  - psql -c "CREATE USER foo WITH PASSWORD 'bar';" -U postgres
```
Comment (adex001) [inline .travis.yml:11]: why are you creating a user for travis?

File: server/tests/users.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../app';
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('POST API/V1/AUTH/SIGNUP /', () => {
```
Comment (adex001) [inline server/tests/users.test.js:8]: This is not descriptive

**Review by adex001 — CHANGES_REQUESTED** [review]

**Pull request merged**